modules/oe_subscriptions_anonymous/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_subscriptions_anonymous\Form;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Entity\Element\EntityAutocomplete;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * The form for the configuration of the Anonymous subscription module.
+ */
+class SettingsForm extends ConfigFormBase {
+
+  /**
+   * Name of the config being edited.
+   */
+  const CONFIG_NAME = 'oe_subscriptions_anonymous.settings';
+
+  /**
+   * EntityTypeManager.
+   *
+   * @var Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'oe_subscriptions_anonymous_settings_form';
+  }
+
+  /**
+   * Constructs a settings form.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    // Saved config.
+    $url = $this->config(static::CONFIG_NAME)->get('terms_url');
+    // Link field.
+    $form['terms_url'] = [
+      '#type' => 'entity_autocomplete',
+      '#target_type' => 'node',
+      '#title' => $this->t('Terms page URL'),
+      '#default_value' => !empty($url) ? $this->getUriAsDisplayableString($url) : '',
+      '#description' => $this->t('The URL to the terms and conditions page.'),
+      '#required' => TRUE,
+      '#element_validate' => [
+        [
+          'Drupal\link\Plugin\Field\FieldWidget\LinkWidget',
+          'validateUriElement',
+        ],
+      ],
+      '#attributes' => [
+        'data-autocomplete-first-character-blacklist' => '/#?',
+      ],
+      '#process_default_value' => FALSE,
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $this->config(static::CONFIG_NAME)
+      ->set('terms_url', $form_state->getValue('terms_url'))
+      ->save();
+
+    Cache::invalidateTags($this->config(static::CONFIG_NAME)->getCacheTags());
```

Comment:
Not totally sure about this change, I think this shouldn't be necessary but another opinion would be great